crates/core/tedge/src/cli/certificate/renew.rs

```diff
@@ -0,0 +1,78 @@
+use super::error::CertError;
+use crate::command::Command;
+use camino::Utf8PathBuf;
+use certificate::KeyCertPair;
+use certificate::NewCertificateConfig;
+use certificate::PemCertificate;
+use std::fs::File;
+use std::fs::OpenOptions;
+use std::io::prelude::*;
+use std::path::Path;
+use tedge_utils::paths::set_permission;
+use tedge_utils::paths::validate_parent_dir_exists;
+
+pub struct RenewCertCmd {
+    /// The path where the device certificate will be stored
+    pub cert_path: Utf8PathBuf,
+}
+
+impl Command for RenewCertCmd {
+    fn description(&self) -> String {
+        "Renew a self-signed certificate for the device.".into()
+    }
+
+    fn execute(&self) -> anyhow::Result<()> {
+        let config = NewCertificateConfig::default();
+        self.renew_test_certificate(&config)?;
+        eprintln!("Certificate was successfully renewed, for un-interrupted service, the certificate has to be uploaded to the cloud");
+        Ok(())
+    }
+}
+
+impl RenewCertCmd {
+    fn renew_test_certificate(&self, config: &NewCertificateConfig) -> Result<(), CertError> {
+        validate_parent_dir_exists(&self.cert_path).map_err(CertError::CertPathError)?;
+
+        // get the ID from the certificate
+        let cert = KeyCertPair::new_selfsigned_certificate(config, &self.device_id()?)?;
```

Comment:
This doesn't work as this method create a certificate for a *new* private key that is not persisted.